internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -250,17 +250,28 @@ func convertEFSPermissions(input map[string]manifest.Volume) ([]*template.EFSPer
 	}
 	output := []*template.EFSPermission{}
 	for _, volume := range input {
-		// Write defaults to false
+		if volume.EFS == nil {
+			// Return nil if there is only one volume and no EFS configuration; otherwise continue.
+			if len(input) == 1 {
+				return nil, nil
+			}
```

Comment:
Maybe we don't need to exit early but just add `EFSPermission` for each volume config? So that if the `input` length is 1 and the only `EFS` setting is `nil` we will continue and just return an empty slice (which is equivalent to `nil`).